File: apps/mobile/lib/models/meeting_point.dart
enum MeetingPointType { park, cafe, landmark }

class MeetingPoint {
  final String id;
  final String name;
  final MeetingPointType type;
  final String address;
  final String neighborhood;
  final String? description;
  final String? photoUrl;
  final String? mapsUrl;

  const MeetingPoint({
    required this.id,
    required this.name,
    required this.type,
    required this.address,
    required this.neighborhood,
    this.description,
    this.photoUrl,
    this.mapsUrl,
  });

  String get typeLabel => switch (type) {
        MeetingPointType.park => 'Parc',
        MeetingPointType.cafe => 'Café',
        MeetingPointType.landmark => 'Point de repère',
      };

  String get typeEmoji => switch (type) {
        MeetingPointType.park => '🌳',
        MeetingPointType.cafe => '☕',
        MeetingPointType.landmark => '📍',
      };
}

File: apps/web/src/lib/types/meeting-point.ts
export type MeetingPointType = "park" | "cafe" | "landmark";

export interface MeetingPoint {
  id: string;
  name: string;
  type: MeetingPointType;
  address: string;
  neighborhood: string;
  description?: string;
  photoUrl?: string;
  mapsUrl?: string;
}

const typeLabels: Record<MeetingPointType, string> = {
  park: "Parc",
  cafe: "Café",
  landmark: "Point de repère",
};

const typeEmojis: Record<MeetingPointType, string> = {
  park: "🌳",
  cafe: "☕",
  landmark: "📍",
};

export function getMeetingPointLabel(type: MeetingPointType): string {
  return typeLabels[type];
}

export function getMeetingPointEmoji(type: MeetingPointType): string {
  return typeEmojis[type];
}
